psh/wget/wget.c

```diff
@@ -0,0 +1,485 @@
+/*
+ * Phoenix-RTOS
+ *
+ * wget - downloads a file using HTTP
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <ctype.h>
+#include <errno.h>
+#include <netdb.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <sys/minmax.h>
+#include <sys/time.h>
+
+#include "../psh.h"
+
+
+static struct {
+	char buf[4096];
+	char *readptr;
+	int len;
+	FILE *outfile;
+	const char *path;
+	char *host;
+	const char *filename;
+	int debug;
+	int socket;
+} common;
+
+
+void psh_wgetinfo(void)
+{
+	printf("downloads a file using http");
+}
+
+
+static void wget_help(void)
+{
+	printf("Usage: wget [options] ... URL\n");
+	printf("Options\n");
+	printf("  -h:  prints help\n");
+	printf("  -O:  output file\n");
+}
+
+
+static int wget_parseUrl(const char *url)
+{
+	const char httpProto[] = "http://";
```

Comment:
```suggestion
	static const char httpProto[] = "http://";
```

Reply:
I don't see any reason for this change.

Reply:
Small stack usage reduction and performance boost - static won't need place and init on stack